istioctl/pkg/cli/context.go

```diff
@@ -103,6 +115,29 @@ func (i *instance) CLIClient() (kube.CLIClient, error) {
 	return i.CLIClientWithRevision("")
 }
 
+func (i *instance) CLIClientsForContexts(contexts []string) ([]kube.CLIClient, error) {
+	rawConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(kube.ConfigLoadingRules(*i.kubeconfig), nil).RawConfig()
+	if err != nil {
+		return nil, fmt.Errorf("getting raw kubeconfig from file %q: %v", *i.kubeconfig, err)
+	}
+	impersonateConfig := i.getImpersonateConfig()
+	rawConfigContexts := maps.Keys(rawConfig.Contexts)
+	for _, c := range contexts {
+		if !slices.Contains(rawConfigContexts, c) {
+			return nil, fmt.Errorf("context %q not found", c)
+		}
+	}
+	var clients []kube.CLIClient
+	for _, contextName := range contexts {
+		c, err := newKubeClientWithRevision(*i.kubeconfig, contextName, "", impersonateConfig)
```

Comment:
Other client initializations are kept in the context to avoid reinitialization.

Reply:
I'm not sure I fully understand the problem here. Which context are you referring to? Could you perhaps point to an example of what you mean (to help me identify the issue with this code)?

Reply:
If you look at how `CliClient()` is initialized, it stores the client in `CliContext`, which is intentional to avoid reinitialization if the function is called twice.

Reply:
Ohh I see what you mean. Added a cache for remote clusters too. PTAL